### PR TITLE
Fixes to make swift project work correctly and fixes to ninja execution path.

### PR DESCRIPTION
**Pull request by onhachoe**
This includes 4 commits

Compile swift before any other clang invocation
 - Swift compilation first before objc compilation.
 - Fixes issue address on #237 

Slight refactoring of PhaseInvocation resolving step
 - as title

Refactor builtin tool resolving in NinjaExecutor
 - Fixes to make symlinked xcbuild work properly

Refactor auxiliary file creation path for ninja
 - Auxiliary file handling improvements.

Add Swift Standard Library tool xcspec
 - swift-stdlib-tool xcspec to copy over swift libraries

### CODE REVIEW

File: Specifications/Tool/com.apple.build-tools.swift-stdlib-tool.xcspec [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            Type = Enumeration;
+            Values = (
+                none,
+                verbose,
+            );
+            DefaultValue = "verbose";
+            CommandLineArgs = {
+                none = ();
+                verbose = (
+                    "--verbose",
```
Comment (modocache) [inline Specifications/Tool/com.apple.build-tools.swift-stdlib-tool.xcspec]: I'm not super familiar with xcspec syntax, so I apologize if this is already true here, but keep in mind `swift-stdlib-tool` has two verbosity settings: `--verbose` (verbose output) and `--verbose --verbose` (debugging output). Does this allow for `--verbose` to be passed twice?

File: Libraries/pbxbuild/Sources/Phase/SwiftResolver.cpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ ShouldBundleSwiftRuntime(Tool::Context const *toolContext, pbxsetting::Environme
      * Need a place to put the runtime libraries. If the product is not a wrapper, then
      * there isn't a frameworks directory to include Swift in.
      */
-    if (productType->isWrapper()) {
+    if (!productType->isWrapper()) {
```
Comment (onhachoe) [inline Libraries/pbxbuild/Sources/Phase/SwiftResolver.cpp]: It did not work at all in the first place... :(

File: Specifications/Tool/com.apple.build-tools.swift-stdlib-tool.xcspec (hunk 1 above)
Comment (onhachoe) [inline Specifications/Tool/com.apple.build-tools.swift-stdlib-tool.xcspec]: oh, I did not realize that was a valid option,
I will update this with support for that.

File: Libraries/xcexecution/Sources/NinjaExecutor.cpp [hunk 3]
```diff
@@ -97,11 +97,11 @@ NinjaDescription(std::string const &description)
 }
 
 static std::string
-NinjaHash(std::string const &input)
+NinjaHashInternal(const char *data, size_t size)
```
Comment (sas) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: `char const *data`, I think.

File: Libraries/xcexecution/Sources/NinjaExecutor.cpp [hunk 4]
```diff
@@ -114,6 +114,12 @@ NinjaHash(std::string const &input)
     return ss.str();
 }
 
+static std::string
+NinjaHash(std::string const &input)
```
Comment (sas) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: What's the point of this? From what I can see, the only caller for these functions actually calls the `Internal` version. Also, keeping the signature the same and not having an extra wrapper seems pretty easy in this case.

File: Libraries/pbxbuild/Sources/Target/Environment.cpp [hunk 5]
```diff
@@ -461,20 +461,23 @@ Create(Build::Environment const &buildEnvironment, Build::Context const &buildCo
...
             // TODO: Apply toolchain override build settings.
             toolchains.push_back(toolchain);
+            effectiveToolchainPaths.push_back(toolchain->path());
         }
     }
 
+    environment.insertFront(pbxsetting::Level({
+        /* At the target level and below, the SDKROOT changes to always be a SDK path. */
+        pbxsetting::Setting::Create("SDKROOT", sdk->path()),
+        pbxsetting::Setting::Create("EFFECTIVE_TOOLCHAINS_DIRS", pbxsetting::Type::FormatList(effectiveToolchainPaths)),
```
Comment (sas) [inline Libraries/pbxbuild/Sources/Target/Environment.cpp]: Does this just insert an empty variable?

Comment (onhachoe) [inline Libraries/pbxbuild/Sources/Target/Environment.cpp]: This is being used by xcspec Specifications/Tool/com.apple.build-tools.swift-stdlib-tool.xcspec

File: Libraries/xcexecution/Sources/NinjaExecutor.cpp (hunk 3 above)
Comment (onhachoe) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: I think they are the same, but okay, I think that is preferable

File: Libraries/xcexecution/Sources/NinjaExecutor.cpp (hunk 4 above)
Comment (onhachoe) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: I might have been lazy with this,
Will change callers' signatures to solve this.

**Pull request merged**